lib/sqlalchemy/ext/orderinglist.py

```diff
@@ -330,13 +336,13 @@ def _order_entity(self, index, entity, reorder=True):
         if have is not None and not reorder:
             return
 
-        should_be = self.ordering_func(index, self)
+        should_be = self.ordering_func(index_to_int(index), self)
```

Comment:
is there any test added for what this code change is providing?   I need to know why this code change is necessary from the point of view of a unit test

Reply:
`index` is always a `SupportsIndex` value, not an integer. I didn't want to change the `ordering_func()` API, which only accepts `int`, so we have to convert the value to an integer, which is what the `operater.index()` function is for.

I'll see about a test for this. It takes creating a custom class implementing `__index__`, something like:

```
class MockIndex:
     def __init__(self, val):
         self.val = val
    def __index__(self):
        return val
```